SU2_CFD/include/CMarkerProfileReaderFVM.hpp

```diff
@@ -164,4 +176,17 @@ class CMarkerProfileReaderFVM {
     return profileData[val_iProfile];
   }
   
+   /*!
+   * \brief Get the data for the specific column if interpolation being done.
+   * \param[in] val_iProfile - current profile index.
+   * \param[in] iCol - the column whose data is required
+   * \returns the specific column data.
+   */
+  inline vector<su2double> &GetColumnForProfile(int val_iProfile, unsigned short iCol) {
+    ColumnData.resize(numberOfRowsInProfile[val_iProfile]);
+  for (unsigned long iRow = 0; iRow < numberOfRowsInProfile[val_iProfile]; iRow++)
+    ColumnData[iRow]=profileData[val_iProfile][iRow*numberOfColumnsInProfile[val_iProfile]+iCol];
+  return ColumnData;
+  }
+
```

Comment:
Here I think this should return by value, like this you allow the outside world to have a non-const reference to the class member data, which is not desired.
```
  inline vector<su2double> GetColumnForProfile(int val_iProfile, unsigned short iCol) const {
    auto nRow = numberOfRowsInProfile[val_iProfile];
    auto nCol = numberOfColumnsInProfile[val_iProfile];
    vector<su2double> ColumnData(nRow);
    for (unsigned long iRow = 0; iRow < nRow; iRow++)
      ColumnData[iRow]=profileData[val_iProfile][iRow*nCol+iCol];
    return ColumnData;
  }
```
(and delete ColumnData from the member variables) (returning by value in C++11 is not bad when the object is local to the function)

Reply:
done